cheatsheets/Browser_Extension_Vulnerabilities_Cheat_Sheet.md

```diff
@@ -1,43 +1,217 @@
 
-## 7. Third-Party Dependencies
+3. Cross-Site Scripting (XSS)
 
-An extension relying on outdated third-party libraries may become vulnerable if those libraries have known security flaws that attackers can exploit.
+🚨 Vulnerability
+If user input is not properly sanitized, attackers can inject malicious scripts into web pages, potentially stealing user data or performing unauthorized actions.
 
-## 8. Lack of Content Security Policy (CSP)
+🔍 Example
 
-Without a strong CSP, attackers can inject untrusted content, increasing the risk of XSS and other attacks that manipulate the extension’s behavior.
+javascript
+Copy code
+let userInput = document.getElementById('input').value;
+document.getElementById('output').innerHTML = userInput; // No sanitization
+🔒 Mitigation
 
-## 9. Insecure Storage
+Implement Content Security Policy (CSP) to block inline scripts.
+Use libraries like DOMPurify to sanitize user input before displaying it.
+Avoid using innerHTML and instead use textContent to prevent execution of injected scripts.
 
-If an attacker gains access to the local storage, they can easily retrieve sensitive information, such as tokens or user credentials, leading to unauthorized access.
 
-## 10. Insufficient Privacy Controls
+4. Insecure Communication
 
-Users may be unaware of how their data is being collected or used, leading to potential abuse of their information without consent or awareness.
+🚨 Vulnerability
+Some extensions send sensitive data over unsecured HTTP connections, making it vulnerable to interception by attackers.
+
+🔍 Example
+
+javascript
+Copy code
+fetch('http://example.com/api/data');
+🔒 Mitigation
+
+Always use HTTPS for external communications to prevent data theft.
+Where possible, implement certificate pinning to prevent man-in-the-middle (MITM) attacks.
+Validate server responses before processing them to ensure data integrity.
+
+
+5. Code Injection
+
+🚨 Vulnerability
+An extension that dynamically loads scripts from an untrusted source can be exploited to inject and execute malicious code.
+
+🔍 Example
+
+javascript
+Copy code
+let script = document.createElement('script');
+script.src = 'http://example.com/malicious.js';
+document.body.appendChild(script);
+🔒 Mitigation
+
+Use CSP (Content Security Policy) to restrict script sources.
+Avoid using eval() and innerHTML as they can execute malicious code.
+Prefer using extension messaging APIs instead of injecting scripts into web pages.
+
+
+6. Malicious Updates
+
+🚨 Vulnerability
+If an extension fetches updates from an untrusted server, an attacker could push malicious updates to all users.
+
+🔍 Example
+
+javascript
+Copy code
+chrome.runtime.onInstalled.addListener(() => {
+  fetch('http://example.com/update-script.js')
+    .then(response => response.text())
+    .then(eval); // Unsafe!
+});
+🔒 Mitigation
+
+Sign extension updates with digital signatures to ensure authenticity.
+Fetch updates only from a trusted HTTPS server.
```

Comment:
Better yet, don't fetch updates within the extension, and only rely on updates from the extension marketplace. See "Don’t inject or incorporate remote scripts" in the [Firefox docs](https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Security_best_practices)